### PR TITLE
Add integrations standardization table

### PR DESCRIPTION
**Pull request by corey-nm**
This will be the markdown file that documents the features each integration support.

Any official integration should add a column to all tables in this markdown, and add either a ✅ if that feature is supported and a ❌ or blank if the feature is not supported.

This will provide a central way to track what each integration supports. Since this is all in 1 file, minimal changes should be needed if integrations are added in the future or if a new row needs to be added.

### CODE REVIEW

File: src/sparseml/INTEGRATIONS.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Training
```
Comment (corey-nm) [inline src/sparseml/INTEGRATIONS.md:1]: Should i name this file README.md so github auto displays it when you're in the sparseml directory?

**Comment by corey-nm** [comment]
Thanks @rahul-tuli and @KSGulin for helping me fill out the table 😁

File: src/sparseml/INTEGRATIONS.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+| Use standard datasets   | ✅     | ✅              | ✅      | ✅            | ✔️      | ✔️          |
+| train/val/test datasets | ✅     | ✅              | ✅      | ✅            | ✔️      | ✔️          |
+| Auto download datasets  | ✅     | ❌              | ✅      | ✅            | ❌      | ✔️          |
+
+| Checkpoints                           | NM IC | torchvision IC | yolov5 | transformers | yolact | openpifpaf |
+| ------------------------------------- | ----- | -------------- | ------ | ------------ | ------ | ---------- |
+| Checkpoints from original integration | ✅     | ✅              | ✅      | ✅            | ✅      | ✅          |
+| Checkpoints from sparsezoo            | ✅     | ✅              | ✅      | ✅            | ✅      | ✅          |
+| Best checkpoint                       | ✅     | ✅              | ✅      | ✅            | ✅      | ✅          |
+| Best Pruned checkpoint                | ❌     | ❌              | ❌      | ❌            | ❌      | ❌          |
```
Comment (corey-nm) [inline src/sparseml/INTEGRATIONS.md:48]: Add note that "Can only be saved after pruning completes"

File: src/sparseml/INTEGRATIONS.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+Key: 
+- ✅ it is supported explicitly by NM code
+- ✔️ it is supported by the underlying integration
+- ❌ it is not supported
+- ❓ unknown, need to check
+
+# Training
+
+| Base                     | NM IC | torchvision IC | yolov5 | transformers | yolact | openpifpaf |
```
Comment (corey-nm) [inline src/sparseml/INTEGRATIONS.md:9]: Add links to all integrations

File: src/sparseml/INTEGRATIONS.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+| ------------------------ | ----- | -------------- | ------ | ------------ | ------ | ---------- |
+| CLI                      | ✅     | ✅              | ✅      | ✅            | ✅      | ✔️          |
+| API                      | ✅     | ✅              | ✅      | ✅            | ✅      | ✔️          |
+| Dense training           | ✅     | ✅              | ✅      | ✅            | ✅      | ✔️          |
+| Gradient accumulation[0] | ❌     | ✅              | ❓      | ✔️            | ❌      | ❌          |
+| DDP                      | ✅     | ✅              | ❓      | ✔️            | ❌      | ✔️          |
```
Comment (corey-nm) [inline src/sparseml/INTEGRATIONS.md:15]: Add a DP row as well

File: src/sparseml/INTEGRATIONS.md (hunk 1 above)
Comment (bfineran) [reply src/sparseml/INTEGRATIONS.md:7]: not right now, but maybe when we decide on a landing page

File: src/sparseml/TRAINING.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+[0] steps_per_epoch should be `len(data_loader) / gradient_accum_steps`
+
+| Sparsification    | NM IC | torchvision IC | yolov5 | transformers | yolact | openpifpaf |
+| ----------------- | ----- | -------------- | ------ | ------------ | ------ | ---------- |
+| Recipe (optional) | ✅     | ✅              | ✅      | ✅            | ✅      | ✅          |
+| Recipe args       | ✅     | ✅              | ✅      | ✅            | ❌      | ❌          |
+| EMA               | ❌     | ✅              | ✅      | ❌            | ❌      | ❌          |
+| AMP[1]            | ✅     | ✅              | ✅      | ✅            | ✅      | ❌          |
+| Distillation[2]   | ❌     | ❌              | ✅      | ✅            | ❌      | ❌          |
```
Comment (corey-nm) [inline src/sparseml/TRAINING.md:26]: Add a row about calibration dataset for quantization

**Comment by bfineran** [comment]
closing with #1384

**Pull request closed without merging**